### PR TITLE
Allow limited ecc_verify operations when compiled with NO_ASN

### PR DESCRIPTION
**Pull request by danielinux**
HAVE_ECC_VERIFY can be now compiled in combination with NO_ASN to allow limited access to the DSA verification interface

### CODE REVIEW

File: wolfcrypt/src/ecc.c
Comment (dgarske) [inline wolfcrypt/src/ecc.c]: Since we have NO_ASN wrappers below around the sign_hash and verify_hash I am not sure this section is even required.

File: wolfcrypt/src/ecc.c
Comment (dgarske) [inline wolfcrypt/src/ecc.c]: Its probably better to expose these in asn_public.h even with NO_ASN is defined. I suspect we'll need other OID's.

File: wolfcrypt/src/sp_cortexm.c
Comment (dgarske) [inline wolfcrypt/src/sp_cortexm.c:15336]: Make sure Sean has this updated in his SP scripts. https://github.com/wolfSSL/scripts/pull/24

**Review by dgarske — CHANGES_REQUESTED** [review]
Thanks

File: wolfcrypt/src/ecc.c
Comment (danielinux) [reply wolfcrypt/src/ecc.c]: Removed the section entirely

File: wolfcrypt/src/ecc.c
Comment (danielinux) [reply wolfcrypt/src/ecc.c]: Moved the whole enum Ecc_Sum to asn_public.h

File: wolfcrypt/src/sp_cortexm.c
Comment (danielinux) [reply wolfcrypt/src/sp_cortexm.c:15336]: mailed Sean about it :+1:

**Comment by danielinux** [comment]
I know this PR is still in the wrong branch, but if the approach looks good now, I'll prepare a similar one for master

**Review by dgarske — APPROVED** [review]
@danielinux : Looks great! Go ahead and submit in wolfSSL master. The changes in sp_cortexm depend on https://github.com/wolfSSL/wolfssl/pull/2028 , so you might want to branch off that or wait until that is merged into master.

**Pull request closed without merging**
